subsys/bootloader/bl_storage/bl_storage.c

```diff
@@ -39,6 +39,100 @@ struct counter_collection {
 #define TYPE_COUNTERS 1 /* Type referring to counter collection. */
 #define COUNTER_DESC_VERSION 1 /* Counter description value for firmware version. */
 
+
+#ifdef PM_S0_OFFSET /* partition manager available */
+/* upgradable mcuboot bootloader option, to validate and select which mcuboot slot to boot:
+ * optional trailer at the end of of flash partitions for mcuboot slot0 and slot1
+ * 1) mcuboot erase flash sector for mcuboot_image_trailer
+ * 2) mcuboot writes new image to flash (e.g. "mcumgr image upload")
+ * 3) mcuboot mcuboot_image_trailer with mcuboot_image_trailer_init
+ * 4) device is rebooted via "mcumgr reboot" or external reset
+ * 5) b0 checks if both mcuboot slots have a valid mcuboot_image_trailer.magic, if not normal boot proceeds
+ * 6) b0 changes mcuboot_image_trailer.status from TESTING to BOOTING (if any TESTING)
+ * 7) b0 gives slot 1st priority if its trailer.status=TESTING/ACTIVE
+ * 8) mcuboot changes trailer.status from BOOTING to ACTIVE if confirmed by "mcumgr image confirm"
+ *    mcuboot also mark passive slot INVALID, to avoid booting in that (except fallback)
+ */
```

Comment:
```suggestion
 * NOTE: if any image fails to validate, the other bank will still be attempted as well. Hence
 *        b0 can "fallback" to an image otherwise marked as INACTIVE.
 */
```